### PR TITLE
Related question

### PR DESCRIPTION
**Pull request by Raemon**

### CODE REVIEW

File: packages/lesswrong/components/questions/RelatedQuestionsList.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          key={rel._id} 
+          post={rel.targetPost} 
+          currentUser={currentUser} 
+          index={i}
+      /> )}
+    </div>
+  )
+}
+
+RelatedQuestionsList.propTypes = {
```
Comment (Discordius) [inline packages/lesswrong/components/questions/RelatedQuestionsList.jsx]: Get rid of unused propTypes

File: packages/lesswrong/lib/collections/posts/schema.js [hunk 2]
```diff
@@ -379,6 +381,79 @@ const schema = {
...
+    hidden: true,
+  },
+
+  sourcePostRelations: resolverOnlyField({
+    type: Array,
+    graphQLtype: '[PostRelation!]',
+    viewableBy: ['guests'],
+    resolver: async (post, args, { Posts }) => {
+      const postRelations = await PostRelations.find({targetPostId: post._id}).fetch()
+
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/schema.js]: Get rid of whitespace

File: packages/lesswrong/lib/collections/posts/schema.js [hunk 3]
```diff
@@ -379,6 +381,79 @@ const schema = {
...
+    viewableBy: ['guests'],
+    resolver: async (post, args, { Posts }) => {
+      const postRelations = await PostRelations.find({targetPostId: post._id}).fetch()
+      
+      return postRelations
+    }
+  }),
+
+  'sourcePostRelations.$': {
+    type: String,
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/schema.js]: Remove this whole field

File: packages/lesswrong/lib/collections/posts/schema.js [hunk 4]
```diff
@@ -379,6 +381,79 @@ const schema = {
...
+          $replaceRoot: {
+            newRoot: "$relatedQuestions"
+          }
+        }
+     ]).toArray()
+     if (!postRelations || postRelations.length < 1) return []
+     return postRelations
+    }
+  }),
+  'targetPostRelations.$': {
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/schema.js]: Remove this whole field

File: packages/lesswrong/lib/collections/posts/views.js [hunk 5]
```diff
@@ -75,6 +79,7 @@ export function augmentForDefaultView(indexFields)
 Posts.addView("userPosts", terms => ({
   selector: { 
     userId: viewFieldAllowAny,
+    hiddenRelatedQuestion: null,
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/views.js]: Replace with viewFieldAllowAny

File: packages/lesswrong/lib/collections/posts/schema.js (hunk 4 above)
Comment (Raemon) [reply packages/lesswrong/lib/collections/posts/schema.js]: No

File: packages/lesswrong/components/editor/EditorFormComponent.jsx [hunk 6]
```diff
@@ -392,8 +392,8 @@ class EditorFormComponent extends Component {
     // or as tall as a post editor) needs to be applied deeper in the tree, for
     // the draftJS editor; if we apply it to our wrapper div, it'll look right
     // but most of it won't be clickable.
-    const heightClass = commentStyles ? classes.commentEditorHeight : classes.postEditorHeight;
 
+    const heightClass = (commentStyles || document.question) ? classes.commentEditorHeight : classes.postEditorHeight;
```
Comment (Raemon) [inline packages/lesswrong/components/editor/EditorFormComponent.jsx]: should this still say "commentEditorHeight" or should we switch to a more generic thing like "Small editor height" and/or differentiate it in a different way?

File: packages/lesswrong/components/posts/PostsItemTooltip.jsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+const getPostCategory = (post) => {
+  const postOrQuestion = post.question ? "Question" : "Post"
+
+  if (post.af) return `Alignment Forum ${postOrQuestion}`
+  if (post.meta) return `Meta ${postOrQuestion}`
+  if (post.curatedDate) return `Curated ${postOrQuestion}`
+  if (post.frontpageDate) return `Frontpage ${postOrQuestion}`
+  if (post.isEvent) return `Event`
+  return post.question ? `Question` : `Personal Blogpost`
```
Comment (Raemon) [inline packages/lesswrong/components/posts/PostsItemTooltip.jsx]: not sure about "Question"/"Personal Bloggpost" thing.

File: packages/lesswrong/components/questions/NewAnswerCommentQuestionForm.jsx [hunk 8]
```diff
@@ -93,6 +93,8 @@ class NewAnswerCommentQuestionForm extends PureComponent {
           post={post}
           type="comment"
         />
+      case "question":
```
Comment (Raemon) [inline packages/lesswrong/components/questions/NewAnswerCommentQuestionForm.jsx]: Should we take the opportunity to also improve our answer/comment ontology? 

We had noticed at least some annoyance at people posting answers that should have been comments, so I think we should at least reverse the defaults there. 

We'd talked about "contribution" and "synthesis." Is that concept within spitting distance of done, enough that we can change the names here?

File: packages/lesswrong/components/questions/NewAnswerCommentQuestionForm.jsx [hunk 9]
```diff
@@ -112,17 +114,18 @@ class NewAnswerCommentQuestionForm extends PureComponent {
                   New Answer
```
Comment (Raemon) [inline packages/lesswrong/components/questions/NewAnswerCommentQuestionForm.jsx]: Should we take the opportunity to also improve our answer/comment ontology? 

We had noticed at least some annoyance at people posting answers that should have been comments, so I think we should at least reverse the defaults there. 

We'd talked about "contribution" and "synthesis." Is that concept within spitting distance of done, enough that we can change the names here?

File: packages/lesswrong/components/questions/NewRelatedQuestionForm.jsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  const { post, classes, flash, currentUser, refetch } = props
+  const { SubmitToFrontpageCheckbox, PostSubmit } = Components
+
+  const QuestionSubmit = (props) => {
+    return <div className={classes.formSubmit}>
+      <SubmitToFrontpageCheckbox {...props}/>
+      <PostSubmit {...props} />
+    </div>
+  }
+  if (!currentUser) return null
```
Comment (Raemon) [inline packages/lesswrong/components/questions/NewRelatedQuestionForm.jsx]: flagging that this currently behaves different from other comment boxes after Jim's update. How much work is it to apply jim's update here?

File: packages/lesswrong/components/questions/NewRelatedQuestionForm.jsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+  return (
+    <div className={classes.root}>
+      <Components.WrappedSmartForm
+        collection={Posts}
+        fields={['title', 'contents', 'body', 'question', 'draft', 'submitToFrontpage', 'hiddenRelatedQuestion', 'originalPostRelationSourceId']}
```
Comment (Raemon) [inline packages/lesswrong/components/questions/NewRelatedQuestionForm.jsx]: do we need both contents and 'body'?

File: packages/lesswrong/lib/collections/posts/views.js [hunk 12]
```diff
@@ -75,6 +79,7 @@ export function augmentForDefaultView(indexFields)
 Posts.addView("userPosts", terms => ({
   selector: { 
     userId: viewFieldAllowAny,
+    hiddenRelatedQuestion: viewFieldAllowAny,
```
Comment (Raemon) [inline packages/lesswrong/lib/collections/posts/views.js]: does this need an index update?

Comment (Raemon) [reply packages/lesswrong/lib/collections/posts/views.js]: I guess not because the default index already includes it?

**Comment by Raemon** [comment]
Things that this PR doesn't do, which it probably shouldn't, but which we might want to do in the near future:

– create a checkbox to show relatedQuestions on the Questions page
– at least move make 'comment' the default option on a question, and possibly change it's name to "contribution", and potentially also change the name "answer" to "synthesis", although that gets messier.

**Comment by Raemon** [comment]
Note that if we're going to merge this PR, we need to also update posts to include "hiddenRelatedQuestion: false"

File: packages/lesswrong/components/comments/CommentsItem/BanUserFromPostMenuItem.jsx [hunk 13]
```diff
@@ -49,7 +49,7 @@ BanUserFromPostMenuItem.propTypes = {
 
 const withEditOptions = {
   collection: Posts,
-  fragmentName: 'PostsPage',
+  fragmentName: 'PostsEdit',
```
Comment (Raemon) [inline packages/lesswrong/components/comments/CommentsItem/BanUserFromPostMenuItem.jsx]: is this still necessary?

**Comment by Raemon** [comment]
Tests to run:

– create normal post, and the EditTitle component renders normally
– edit a normal post, and you can edit it properly (testing fragment updates)


– create (normal) question post, and...
     -    EditTitle component renders in question form (smaller)
     -    editor is smaller
     -    posting works normally

– create (related) question post, and...
     -    EditTitle component renders in question form (smaller)
     -    editor is smaller
     -    after posting...
               - EditTitle is cleared
               - main content is cleared
               - question immediately appears in the "Related Questions" section
               - question does NOT appear on frontpage, or questions page, or allPosts page
               - question DOES appear on user profile
               - question DOES appear on allPostsPage if set to "questions"
               - question DOES appear on allPosts, questions and frontpage, if "includeRelatedQuestions=true" is in url, or "filter=questions" in url

– PostsItemTooltip...
     -    appears properly (since it was refactored)
     -    renders "AI Alignment Forum Post" if on an AF post
     -    renders "AI Alignment Forum Question" if on an AF question
     -    renders "Frontpage Question" for frontpage questions
     -    renders "Question" for personal blogpost questions
     -    renders "Personal Blogpost" for personal blogpost

– Subquestions
     -    A given question will show questions 2 nodes deep (but not 3)
     -    A question will display it's parent related question, labeled in a distinguishable way [FAILING]

**Comment by Raemon** [comment]
I added the parent/subquestion headings back in because it was confusing. It looks a little janky, but I think better to push it as is and improve it later.

image

File: packages/lesswrong/components/questions/NewRelatedQuestionForm.jsx (hunk 11 above)
Comment (Discordius) [reply packages/lesswrong/components/questions/NewRelatedQuestionForm.jsx]: No

**Pull request merged**